nrf_wifi/os_if/src/osal.c

```diff
@@ -11,206 +11,171 @@
 #include "osal_api.h"
 #include "osal_ops.h"
 
-struct nrf_wifi_osal_priv *nrf_wifi_osal_init(void)
-{
-	struct nrf_wifi_osal_priv *opriv = NULL;
-	const struct nrf_wifi_osal_ops *ops = NULL;
-
-	ops = get_os_ops();
-
-	opriv = ops->mem_zalloc(sizeof(struct nrf_wifi_osal_priv));
+struct nrf_wifi_osal_ops *os_ops;
 
-	if (!opriv) {
-		goto out;
-	}
-
-	opriv->ops = ops;
-
-out:
-	return opriv;
+void nrf_wifi_osal_init(struct nrf_wifi_osal_ops *ops)
+{
+	os_ops = ops;
 }
 
 
-void nrf_wifi_osal_deinit(struct nrf_wifi_osal_priv *opriv)
+void nrf_wifi_osal_deinit(void)
 {
-	const struct nrf_wifi_osal_ops *ops = NULL;
-
-	ops = opriv->ops;
-
-	ops->mem_free(opriv);
+	os_ops = NULL;
```

Comment:
As this is a global variable, we need to protect this else during de-init it might crash. Ideally RCU is suitable for this, but in Zephyr we can stick with a mutex.

Reply:
This is expected to be called only during the driver deinit, so shouldn't be an issue?

Reply:
Yeah, it's the last step of de-init, so, there won't be callers using `os_ops` I guess.